runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java

```diff
@@ -141,26 +164,49 @@ private void translateKinesisInput(
     RunnerApi.PTransform pTransform = pipeline.getComponents().getTransformsOrThrow(id);
 
     String stream;
+    FlinkLyftKinesisConsumer<WindowedValue<byte[]>> source;
+
     Properties properties = new Properties();
     ObjectMapper mapper = new ObjectMapper();
     try {
-      Map<String, Object> params =
-          mapper.readValue(pTransform.getSpec().getPayload().toByteArray(), Map.class);
+      JsonNode params = mapper.readTree(pTransform.getSpec().getPayload().toByteArray());
 
       Preconditions.checkNotNull(
-          stream = (String) params.get("stream"), "'stream' needs to be set");
-      Map<?, ?> consumerProps = (Map) params.get("properties");
+          stream = params.path("stream").textValue(), "'stream' needs to be set");
+
+      Map<?, ?> consumerProps = mapper.convertValue(params.path("properties"), Map.class);
       Preconditions.checkNotNull(consumerProps, "'properties' need to be set");
       properties.putAll(consumerProps);
+
+      String encoding = BYTES_ENCODING;
+      if (params.hasNonNull("encoding")) {
+        encoding = params.get("encoding").asText();
+      }
+
+      switch (encoding) {
+        case BYTES_ENCODING:
+          source =
+              FlinkLyftKinesisConsumer.create(
+                  stream, new KinesisByteArrayWindowedValueSchema(), properties);
+          break;
+        case LYFT_KINESIS_EVENT_ENCODING:
+          source = FlinkLyftKinesisConsumer.create(stream, new LyftBeamKinesisSchema(), properties);
+          source.setPeriodicWatermarkAssigner(new WindowedTimestampExtractor<>(Time.seconds(1)));
```

Comment:
maxOutOfOrderness seems a tad aggressive. Should this be a parameter?

Reply:
Do you have a suggestion for the default? Does it make sense to have one? Regardless will make it configurable.

Reply:
It really depends on the application. If I had to pick a default, then maybe 5 seconds. See: https://ci.apache.org/projects/flink/flink-docs-release-1.6/dev/event_timestamps_watermarks.html#with-periodic-watermarks

Reply:
It may be necessary to have control over idle shard handling also: https://github.com/lyft/streamingplatform/blob/master/streamingplatform-kinesis/src/main/java/com/lyft/streamingplatform/flink/FlinkLyftKinesisConsumer.java#L27